Firestore/Example/Tests/Remote/FSTSerializerBetaTests.mm

```diff
@@ -382,6 +382,23 @@ - (void)testEncodesTransformMutation {
   [self assertRoundTripForMutation:mutation proto:proto];
 }
 
+- (void)testEncodesArrayTransformMutations {
+  FSTTransformMutation *mutation = FSTTestTransformMutation(@"docs/1", @{
+    @"a" : [FIRFieldValue fieldValueForArrayUnion:@[ @"a", @2 ]],
+    @"bar.baz" : [FIRFieldValue fieldValueForArrayRemove:@[
+      @{ @"x" : @1 }
+    ]]
+  });
+  GCFSWrite *proto = [GCFSWrite message];
+  proto.transform = [GCFSDocumentTransform message];
+  proto.transform.document = [self.serializer encodedDocumentKey:mutation.key];
+  proto.transform.fieldTransformsArray =
+      [self.serializer encodedFieldTransforms:mutation.fieldTransforms];
```

Comment:
This is a pretty weak assertion. If `encodedFieldTransforms:` returned `nil` the check on the encode pass wouldn't catch it.

The only thing catching this is the decode pass has to reconstitute the original, which while something is a pretty indirect check.